src/Microsoft.TemplateEngine.Cli/TemplateListResolutionResult.cs

```diff
@@ -256,5 +256,55 @@ public bool UsingContextMatches
                 return _usingContextMatches;
             }
         }
+
+        public bool IsTemplateAmbiguous { get; private set; }
+
```

Comment:
I'd consider changing IsTemplateAmbiguous and IsNoTemplatedMatchedState getters to an Ensure() type pattern, so that ComputeContextBasedAndOtherPartialMatches() doesn't need to be explicitly called to use these properties.

Reply:
Same idea would need to apply to ContextProblemMatchGroups and RemainingPartialMatchGroups

Reply:
I made the method private and am now calling it as the last line of the contructor. I also fixed a null reference that was coming up when a template had a null groupIdentity in that logic